### PR TITLE
Deposit Command: /deposit [all] [amount]

### PR DESCRIPTION
**Pull request by semklauke**
## Content:
1. "Reworked" the deposit command.
    - `deposit` now just deposits the items in your hand.
    - `deposit all [int: amount]` has the old behavior, but can also be capped by `amount`
2. Added a config option for currency pre- & suffix. Only used it deposit command in this commit

## Discussion
When you have multiple currency items configured, but only want to deposit a specifc or not all of them, you would have to clear your inventory first. With the hand feature, you can easily control what you deposit.
Maybe you could swap the command structure: `deposit` stays the same and the hand feature move to 
`deposit hand`

---

Hey,

I added some feature on my personal fork. Some might also work for the main repo. So I'm testing the water here.
If you like the ideas I have, I would add some more PRs.

Open to discussion/changes/feedback :)

### CODE REVIEW

**Comment by Ranchosm335** [comment]
Deposit only certain currency matches  region

**Comment by semklauke** [comment]
@Ranchosm335 sorry, I don’t get what you mean. What region ?

**Comment by Ranchosm335** [comment]
https://jekyllrb.com/docs/configuration/front-matter-defaults/

File: README.md [hunk 1]
```diff
@@ -2,7 +2,8 @@ Basic economy for the fabric modloader using diamonds as currency. Uses the "/di
 
 - balance [Optional: player] - tells user how much money the player has
```
Comment (Ranchosm335) [inline README.md:3]: I not aware of amount exchanged at instance the balance is calculated is subject to change values

File: src/main/java/com/gmail/sneakdevs/diamondeconomy/command/DepositCommand.java [hunk 2]
```diff
@@ -3,40 +3,111 @@
 import com.gmail.sneakdevs.diamondeconomy.DiamondUtils;
```
Comment (Ranchosm335) [inline src/main/java/com/gmail/sneakdevs/diamondeconomy/command/DepositCommand.java:3]: Shall the value embed in code written with my data portfolio

**Review by Ranchosm335 — COMMENTED** [review]
Approve. Account balance Owners Investment in 1BTC=current value amount multiples by the volume of shares in null foreign currency 
Losses and Gains statement attached value is assessed by the time stamped statement Date.Hour.Minute.Seconds. GMT

**Comment by Ranchosm335** [comment]
{
  "name": "BTC.main",
  "height": 865493,
  "hash": "0000000000000000000143a112c5ab741ec6e95b6c80f9834199efe2154c972b",
  "time": "2024-10-13T16:25:49.713998704Z",
  "latest_url": "https://api.blockcypher.com/v1/btc/main/blocks/0000000000000000000143a112c5ab741ec6e95b6c80f9834199efe2154c972b",
  "previous_hash": "000000000000000000019ebb16c408aa943419a77bdab3942da4b17520fc9dae",
  "previous_url": "https://api.blockcypher.com/v1/btc/main/blocks/000000000000000000019ebb16c408aa943419a77bdab3942da4b17520fc9dae",
  "peer_count": 324,
  "unconfirmed_count": 26112,
  "high_fee_per_kb": 28089,
  "medium_fee_per_kb": 13721,
  "low_fee_per_kb": 7758,
  "last_fork_height": 863888,
  "last_fork_hash": "000000000000000000022336ca639c1483ce4bfc33f07909659c1be89d133238"
}

**Review by Ranchosm335 — COMMENTED** [review]
Accounts receivable for deposit of deposited finds

**Pull request merged**